src/reconcile/rec_write.c

```diff
@@ -2341,6 +2355,7 @@ __wt_rec_cell_build_ovfl(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_KV *k
         dsk = tmp->mem;
         memset(dsk, 0, WT_PAGE_HEADER_SIZE);
         dsk->type = WT_PAGE_OVFL;
+        dsk->write_gen = ++btree->write_gen;
```

Comment:
Another odd thing is that we are building a cell not a page here.

Reply:
It is a cell on the data page and a normal page is written with an overflow record.